src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final double MAX_SIZE = 10.0;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
```

Comment:
Consider using a constant for the number of figure types (5 in this case) instead of hardcoding it in the `nextInt` method. This would make the code more maintainable and self-explanatory.